datafusion/functions/src/math/round.rs

```diff
@@ -109,6 +109,9 @@ pub fn round(args: &[ArrayRef]) -> Result<ArrayRef> {
 
     if args.len() == 2 {
         decimal_places = ColumnarValue::Array(args[1].clone());
+        // if cast fails, we return error instead of panic. Better solution would to update
+        // function signature and instead error during planning -- to be improved later
+        _ = decimal_places.cast_to(&Int32, None)?;
```

Comment:
I think you could potentially just check the result of the conversion below too

So instead of the following on line 120:

```rust
                let decimal_places = decimal_places.try_into().unwrap();
``` 

something like this?

```rust
                let decimal_places: i32 = decimal_places.try_into()
                    .map_err(|e| exec_datafusion_err!("Invalid value for decimal places: {decimal_places}: {e}"))?; 

```

Reply:
@alamb sure thing, thx for the suggestion -- I've added it in the recent commit, both for the scalar and array cases. Does it look ok?

Reply:
It looks good -- thanks! 

Now that you are checking the result of converting `decimal_places`I think we can avoid the call to `cast`

Reply:
right, thx 😇 Removed & updated tests match; all seems to behave well